decred/examples/drain_xpriv.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+"""
+Copyright (c) 2020, The Decred developers
+
+This example script will dump all funds located in an xpriv to a specified address.
+"""
+import os
+from urllib.parse import urlunsplit
+
+from base58 import b58decode, b58encode
+from decred import DecredError
+from decred.crypto import crypto
+from decred.crypto.secp256k1.curve import curve as Curve
+from decred.dcr import account, addrlib, nets, rpc, txscript
+from decred.dcr.wire import msgtx, wire
+from decred.util import helpers
+from decred.util.encode import ByteArray
+
+
+SERIALIZED_KEY_LENGTH = 4 + 1 + 4 + 4 + 32 + 33  # 78 bytes
+INTERNAL = 0
+EXTERNAL = 1
+FEE_RATE = 10000
+GAP_LIMIT = 20
+
+
+def cfg(isTestnet):
+    dcrdCfgDir = helpers.appDataDir("dcrd")
+    cfgPath = os.path.join(dcrdCfgDir, "dcrd.conf")
+    if not os.path.isfile(cfgPath):
+        return None
+    cfg = helpers.readINI(cfgPath, ["rpcuser", "rpcpass", "rpccert"])
+    assert "rpcuser" in cfg
+    assert "rpcpass" in cfg
+    if "rpccert" not in cfg:
+        cfg["rpccert"] = os.path.join(dcrdCfgDir, "rpc.cert")
+    if "rpclisten" not in cfg:
+        cfg["rpclisten"] = "localhost:9109"
+    if isTestnet:
+        cfg["rpclisten"] = "localhost:19109"
+    return cfg
+
+
+def decodeExtendedKey(netParams, key):
```

Comment:
Definitely encouraged to modify package code to help with examples. What's good is the code but if not to help with stuff like this? Consider https://github.com/JoeGruffins/tinydecred/compare/dumpxpriv...buck54321:jg?expand=1

Reply:
I agree, but I really wanted to make this script accessible with the release. I will change this in a later pr...